server/src/main/java/org/elasticsearch/action/bulk/BulkRequestParser.java

```diff
@@ -111,34 +108,13 @@ public void parse(
             @Nullable String defaultRouting, @Nullable FetchSourceContext defaultFetchSourceContext,
             @Nullable String defaultPipeline, boolean allowExplicitIndex,
             XContentType xContentType,
-            Consumer<IndexRequest> indexRequestConsumer,
-            Consumer<UpdateRequest> updateRequestConsumer,
-            Consumer<DeleteRequest> deleteRequestConsumer) throws IOException {
-        parse(data, defaultIndex, null, defaultRouting, defaultFetchSourceContext, defaultPipeline, allowExplicitIndex, xContentType,
-                indexRequestConsumer, updateRequestConsumer, deleteRequestConsumer);
-    }
-
-    /**
-     * Parse the provided {@code data} assuming the provided default values. Index requests
-     * will be passed to the {@code indexRequestConsumer}, update requests to the
-     * {@code updateRequestConsumer} and delete requests to the {@code deleteRequestConsumer}.
-     * @deprecated Use {@link #parse(BytesReference, String, String, FetchSourceContext, String, boolean, XContentType,
-     * Consumer, Consumer, Consumer)} instead.
-     */
-    @Deprecated
-    public void parse(
-            BytesReference data, @Nullable String defaultIndex, @Nullable String defaultType,
-            @Nullable String defaultRouting, @Nullable FetchSourceContext defaultFetchSourceContext,
-            @Nullable String defaultPipeline, boolean allowExplicitIndex,
-            XContentType xContentType,
-            Consumer<IndexRequest> indexRequestConsumer,
+            BiConsumer<IndexRequest, String> indexRequestConsumer,
```

Comment:
This is a slightly hacky way of getting around the fact that the Monitoring Bulk action still uses the `_type` parameter in a bulk index line to set a `type` field.  This action is slated for deprecation and removal at some point in the future; until then, seeing as the BulkParser no longer sets the type on its IndexRequest, we need to pass it separately to this consumer.

Reply:
I don't love that we need to maintain types in one place in the bulk action because of the monitoring bulk action. I'd like to try to find some way to avoid this whilst minimising the affect on the monitoring bulk actions API itself. This doesn't need to be solved in this PR but I'd like to try to solve it in the types removal for 8.0 work